### PR TITLE
Image example

### PR DESCRIPTION
**Pull request by JohnMcLear**
Here I'm doing a few things.

1) A really solid example for baking ``img`` elements into core, but without it being exposed.  IE I will bake it in, then before this is merged it will be intentionally removed.
2) Linting and tidy up of the content collector logic.
3) contentcollector test coverage for images

### CODE REVIEW

File: src/static/js/contentcollector.js [hunk 1]
```diff
@@ -704,13 +695,14 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
...
           fixLineNumber(ss);
           fixLineNumber(se);
           linesWrapped++;
           numLinesAfter += newStrings.length;
 
           newStrings.unshift(i, 1);
+          // Still to fix linting issue below.
           lineStrings.splice.apply(lineStrings, newStrings);
           newAttribStrings.unshift(i, 1);
           lineAttribs.splice.apply(lineAttribs, newAttribStrings);
```
Comment (HMarzban) [inline src/static/js/contentcollector.js:708]: Replace these lines:
```js
newStrings.unshift(i, 1);
// Still to fix linting issue below.
lineStrings.splice.apply(lineStrings, newStrings);
newAttribStrings.unshift(i, 1);
lineAttribs.splice.apply(lineAttribs, newAttribStrings);
```
with this two lines:
```js
lineStrings = [...lineStrings, i, 1, ...newStrings];
lineAttribs = [...lineAttribs, i, 1, ...newAttribStrings];
```

File: src/static/js/domline.js [hunk 2]
```diff
@@ -73,20 +73,19 @@ domline.createDomLine = function (nonEmpty, doesWrap, optBrowser, optDocument) {
...
 
-  result.appendSpan = function (txt, cls) {
+  result.appendSpan = (txt, cls) => {
+    // console.log("domline", domline, "txt", txt, "cls", cls)
     let processedMarker = false;
     // Handle lineAttributeMarker, if present
     if (cls.indexOf(lineAttributeMarker) >= 0) {
       let listType = /(?:^| )list:(\S+)/.exec(cls);
       const start = /(?:^| )start:(\S+)/.exec(cls);
+      const img = /(?:^| )img:(\S+)/.exec(cls);
```
Comment (JohnMcLear) [inline src/static/js/domline.js:88]: This logic would be in a plugin!

File: src/static/js/domline.js [hunk 3]
```diff
@@ -97,6 +96,11 @@ domline.createDomLine = function (nonEmpty, doesWrap, optBrowser, optDocument) {
         processedMarker |= modifier.processedMarker;
       });
 
+      if (img) {
```
Comment (JohnMcLear) [inline src/static/js/domline.js:99]: This logic would be in a plugin!

File: src/static/js/linestylefilter.js [hunk 4]
```diff
@@ -92,6 +91,8 @@ linestylefilter.getLineStyleFilter = (lineLength, aline, textAndClassFunc, apool
               classes += ` start:${value}`;
             } else if (linestylefilter.ATTRIB_CLASSES[key]) {
               classes += ` ${linestylefilter.ATTRIB_CLASSES[key]}`;
+            } else if (key === 'img') {
```
Comment (JohnMcLear) [inline src/static/js/linestylefilter.js:94]: This logic would be in a plugin!

File: src/static/js/contentcollector.js [hunk 5]
```diff
@@ -31,106 +31,80 @@ const Changeset = require('./Changeset');
...
-    },
+    optNodeInnerHTML: (n) => n.innerHTML,
   };
 
   const _blockElems = {
     div: 1,
     p: 1,
     pre: 1,
     li: 1,
+    img: 1,
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js:66]: This logic would be in a plugin!

File: src/static/js/contentcollector.js [hunk 6]
```diff
@@ -411,23 +389,36 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
...
           cls,
           node,
         });
-      } else {
-        // THIS SEEMS VERY HACKY! -- Please submit a better fix!
-        delete state.lineAttributes.img;
-      }
-
-      if (tname == 'br') {
+        // Client side processing
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js:413]: This logic would be in a plugin!

File: src/static/js/contentcollector.js [hunk 7]
```diff
@@ -411,23 +389,36 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
...
+        if (node.src) state.lineAttributes.img = node.src;
+        // Server side processing
```
Comment (JohnMcLear) [inline src/static/js/contentcollector.js:415]: This logic would be in a plugin!

File: src/static/js/contentcollector.js (hunk 1 above)
Comment (HMarzban) [reply src/static/js/contentcollector.js:721]: Well, finally, replace these lines with these
```
lineStrings.splice(i, 1, ...newStrings);
lineAttribs.splice(i, 1, ...newAttribStrings);
```

File: src/static/js/contentcollector.js [hunk 8]
```diff
@@ -295,24 +272,24 @@ function makeContentCollector(collectStyles, abrowser, apool, domInterface, clas
         _.map(state.lineAttributes, (value, key) => [key, value])
     );
```
Comment (HMarzban) [inline src/static/js/contentcollector.js:273]: These features can be replaced with es6 version:
```js
const attributes = [
  ['lmkr', '1'],
  ['insertorder', 'first'],
  ..Object.entries(state.lineAttributes)
]
```

**Pull request closed without merging**